pkg/url/url.go

```diff
@@ -899,11 +920,55 @@ func Push(client *occlient.Client, kClient *kclient.Client, parameters PushParam
 					Port:   url.Port,
 					Secure: url.Secure,
 					Kind:   envinfo.ROUTE,
+					Path:   "/",
 				},
 			}
 		}
 	}
 
+	// iterate through endpoints defined in devfile
+	// add the url defination into urlLOCAL if it's not defined in env.yaml
+	if parameters.IsExperimentalModeEnabled && parameters.EndpointMap != nil {
+		for port, endpoint := range parameters.EndpointMap {
+			// should not create URL if Exposure is none or internal
+			if endpoint.Exposure == "none" || endpoint.Exposure == "internal" {
+				continue
+			}
+			exist := false
+			for _, envURL := range urlLOCAL {
+				if envURL.Spec.Port == int(port) {
+					exist = true
+					break
+				}
+			}
+			if !exist {
```

Comment:
Why are we not creating the endpoint URL if the port is used by a env URL?

Reply:
It is because of this https://github.com/openshift/odo/pull/3665#discussion_r468094439

As described is the PR description, 
```
For same port, if urlName in env.yaml is different from devfile.yaml, will use the name from env.yaml; since env.yaml is considered as a "local" configuration, user runs odo create cli to intentionally set the value, therefore env.yaml has precedence over devfile.yaml
```

We have discussed the issue in last Thursday's meeting, for this PR, it is simply to support url creation with endpoint settings defined in devfile. Due to the awkward situation we are in, with both env.yaml and devfile.yaml; we want to implement https://github.com/openshift/odo/issues/3718 before we make more changes on URL creation logics.

Reply:
So this was implemented to support the second case in mentioned here https://github.com/openshift/odo/pull/3665#discussion_r468624308?

Reply:
Yes. If we want to implement the first option, then this means we only allow user to create url using name defined in devfile.yaml, which bring more limitation. Which means, it breaks the current feature, user can no longer create one route and one ingress at the same time using the same port, coz only one URL can be create to match the endpoint name. 
Almost all the integration test need to be changed because of that change, and we need to change those tests back when we implement #3718. 

Some of your suggestions can be made as part of the PR and also is with good reason; the question is, do we want to spend those effort knowing that we are going to work on #3718 as soon as this PR get merged.

Reply:
>  we only allow user to create url using name defined in devfile.yaml, which brings more limitation

I agree but the other case brings confusion described in https://github.com/openshift/odo/pull/3665#discussion_r468605247.

> Which means, it breaks the current feature. ( For example, user can no longer create one route and one ingress at the same time using the same port, coz only one URL can be created to match the endpoint name. )

I didn't get you. How does it break the current feature? They can use `odo url create` and create a route and ingress on the same port if they want. What am I missing?

Reply:
Anyways leave it. Let's fix them in #3718.